static/src/js/components/Map/Map.jsx

```diff
@@ -286,6 +330,59 @@ const Map = ({
   const mapRef = useRef()
   const mapElRef = useRef()
 
+  useEffect(() => {
+    // Set initial base layer visibility
+    worldImageryLayer.setVisible(base.worldImagery)
+    correctedReflectanceLayer.setVisible(base.trueColor)
+    landWaterMapLayer.setVisible(base.landWaterMap)
+
+    // Set initial overlay layer visibility
+    bordersRoadsLayer.setVisible(overlays.referenceFeatures)
+    coastlinesLayer.setVisible(overlays.coastlines)
+    placeLabelsLayer.setVisible(overlays.referenceLabels)
+  }, [])
+
+  const handleLayerChange = ({ id, checked }) => {
+    // Handle base layers
+    if (baseLayerIds.includes(id)) {
+      const newBase = {
+        worldImagery: false,
+        trueColor: false,
+        landWaterMap: false
+      }
+
+      // Set the selected base layer to true in the Redux state
+      if (id === mapLayers.worldImagery) newBase.worldImagery = checked
+      if (id === mapLayers.correctedReflectance) newBase.trueColor = checked
+      if (id === mapLayers.landWaterMap) newBase.landWaterMap = checked
+
+      // Update all base layer visibility
+      Object.keys(baseLayers).forEach((layerId) => {
+        baseLayers[layerId].setVisible(layerId === id && checked)
+      })
+
+      onChangeMap({ base: newBase })
```

Comment:
You don't need two separate calls to onChangeMap, it doesn't matter if the selection was a base layer or overlay layer, both end up calling onChangeMap, move it outside of the conditional like in my example